api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java

```diff
@@ -378,6 +381,72 @@ public void restoreFileVersion(final GSBucketStorage storage, final String path,
         deleteBlob(blob, client, true);
     }
 
+    public void restoreFolder(final GSBucketStorage storage, final String path,
+                              final RestoreFolderVO restoreFolderVO) {
+        final Storage client = gcpClient.buildStorageClient(region);
+        final String bucketName = storage.getPath();
+        cleanDeleteMarkers(client, bucketName, path, restoreFolderVO);
+    }
+
+    private void cleanDeleteMarkers(final Storage client,
+                                    final String bucketName, final String requestPath,
+                                    final RestoreFolderVO restoreFolderVO) {
+        String folderPath = Optional.ofNullable(requestPath).orElse(EMPTY_PREFIX);
+        if (StringUtils.isNotBlank(folderPath)) {
+            folderPath = normalizeFolderPath(requestPath);
+        }
+        final Page<Blob> blobs = client.list(bucketName,
```

Comment:
Maybe it would be better to move this logic into separate method and reuse it into `listItems` methd?

Reply:
Did it, now it looks little bit better